src/main/tut/1.9-function-types.md

```diff
@@ -0,0 +1,143 @@
+```Haskell
+eval :: ((a⇒b) × a) -> b
+```
+```scala
+def eval[`A=>B`, A, B]: ((`A=>B`, A)) => B
```

Comment:
This is incorrect, the following compiles:
```scala
  def eval[`A=>B`, A, B]: ((`A=>B`, A)) => B = ???
  def no = eval[String, Boolean, Double]
```
The literal translation of your code should be
```scala
def eval[`=>`[_, _], A, B]
```
but that's not what the text is saying. `⇒` and `×` are not type parameters there, but concrete types. The text says that `eval` is a morphism associated with the function object, whereas if you add it as a type parameter it means that it'll work `forall` the types of the given shape.